GitCommands/Git/GitCreateTagCmd.cs

```diff
@@ -27,86 +24,104 @@ public static bool CanProvideMessage(this TagOperation operationType)
                 case TagOperation.SignWithSpecificKey:
                     return true;
                 default:
-                    throw new NotSupportedException("Invalid TagOperation: " + operationType);
+                    throw new NotSupportedException($"Invalid tag operation: {operation}");
             }
         }
     }
 
-    public sealed class GitCreateTagArgs
+    public class GitCreateTagArgs
     {
-        public string Revision;
-        public string TagName;
-        public bool Force = false;
-        public TagOperation OperationType = TagOperation.Lightweight;
-        public string TagMessage = "";
-        public string SignKeyId = "";
+        /// <summary>
+        /// Creates arguments for creation of a tag.
+        /// </summary>
+        /// <param name="tagName">Name of tag</param>
+        /// <param name="revision">Commit revision to be tagged</param>
+        /// <param name="operation">The operation to perform on the tag</param>
+        /// <param name="tagMessage">Tag Message</param>
+        /// <param name="signKeyId">Specific Key ID to be used instead of default one</param>
+        /// <param name="force">Force parameter</param>
+        public GitCreateTagArgs(string tagName, string revision, TagOperation operation = TagOperation.Lightweight, string tagMessage = "", string signKeyId = "", bool force = false)
```

Comment:
How does it relate to https://github.com/gitextensions/gitextensions/issues/1748?

Reply:
I don't believe it does.

1. For starters, I have raised it as a problem on the original PR
![image](https://user-images.githubusercontent.com/4403806/31305298-2e3ae360-ab82-11e7-9654-ec0a1dce882a.png)

2. `GitCreateTagArgs` is a data transport object to minimise the number of parameters for the command and not the API. The API is `GitCreateTagCmd.GitCreateTagCmd(GitCreateTagArgs, string)`
If you object requires 5 parameters you have to supply them one way or another.

I locked the args down to avoid possible side-effects.

Reply:
Perhaps I don't understand the intention of #1748 and the accepted solution for the related SO question. @KindDragon could you clarify when #1748 applies and what is the desired solution?

Reply:
Can we not get bogged down on this? I have two PRs blocked by this one, which are probably much more interesting to discuss